Gems/Atom/RPI/Code/Source/RPI.Public/Shader/ShaderVariant.cpp

```diff
@@ -82,25 +78,5 @@ namespace AZ
             }
         }
 
-
-        void ShaderVariant::OnAssetReloaded(Data::Asset<Data::AssetData> asset)
```

Comment:
It's ok to remove because in Shader.cpp the following notification is triggered when a Variant is created or reloaded.
void Shader::OnShaderVariantAssetReady(Data::Asset<ShaderVariantAsset> shaderVariantAsset, bool isError)
It update the ShaderVariant, and triggers the same notification that ShaderVariant::OnAssetReloaded() was doing and it is not redundant because shaderVariant.Init() is now called only once from Shader.cpp